client/my-sites/site-settings/jetpack-credentials/credentials-setup-flow/setup-footer.jsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Gridicon from 'gridicons';
+import Popover from 'components/popover';
+
+class SetupFooter extends Component {
+	componentWillMount() {
+		this.setState( { showPopover: false } );
+	}
+
+	togglePopover = () => this.setState( { showPopover: ! this.state.showPopover } );
+
+	render() {
+		const { translate } = this.props;
+
+		return (
+
+			<CompactCard className="credentials-setup-flow__footer">
+				<a
+					onClick={ this.togglePopover }
+					ref="popoverLink"
```

Comment:
While this is okay, we generally prefer to use the _function syntax_ for `ref`s which keeps things a little tighter in the flow of data and control. Here's an example…

```js
// store the ref on some class property
storePopoverLink = ref => this.popoverLink = ref;

render() {
	…
	ref={ this.storePopoverLink }
	…
	content={ this.popoverLink }
	…
}
```

A couple things here:
	- `this.popoverLink` is _always_ valid; if the component hasn't yet mounted it will simply be `undefined`
	- no more "stringly-typed" references